### PR TITLE
rpcdaemon: handle incomplete request buffers using picohttpparse

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/silkrpc/http/request_parser.cpp [hunk 1]
```diff
@@ -29,6 +29,19 @@
 
 namespace silkworm::rpc::http {
 
+void RequestParser::reset() {
+    if (last_len_) {
+        last_len_ = 0;
+        delete[] buffer_;
```
Comment (canepat) [inline silkworm/silkrpc/http/request_parser.cpp:35]: It is better to use smart pointers to deal with dynamic memory. You can use just `std::vector<char>` as declaration type for your `buffer_`.

File: silkworm/silkrpc/http/request_parser_test.cpp [hunk 2]
```diff
@@ -37,10 +38,10 @@ TEST_CASE("parse", "[silkrpc][http][request_parser]") {
             const auto result{parser.parse(req, buffer.data(), buffer.data() + bytes_read)};
             CHECK(result == RequestParser::ResultType::bad);
         }
-    }
+    }  // namespace silkworm::rpc::http
```
Comment (canepat) [inline silkworm/silkrpc/http/request_parser_test.cpp:41]: I suspect this is a typo because this curly brace does _not_ close the namespace

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/silkrpc/http/request_parser.cpp (hunk 1 above)
Comment (lupin012) [reply silkworm/silkrpc/http/request_parser.cpp:35]: ok

File: silkworm/silkrpc/http/request_parser_test.cpp (hunk 2 above)
Comment (lupin012) [reply silkworm/silkrpc/http/request_parser_test.cpp:41]: ok

**Review by canepat — APPROVED** [review]

**Comment by yperbasis** [comment]
H by.   Night nun is
M  im I’m oI kill io KKK MMO mythe o km look mal i
________________________________

**Pull request merged**